### ISSUE TITLE
devspace render runtime error

### ISSUE THREAD

**Issue by chasebolt** [devspace-cloud/devspace#927]
**What happened?**  
I ran `devspace render` with a config that includes dependencies

**What did you expect to happen instead?**  
The rendered output

**How can we reproduce the bug?** (as minimally and precisely as possible)  
My current `devspace.yaml` contains
```yaml
dependencies:
  - source:
      git: git@github.com:myrepo/k8s-base.git
      subPath: /kube-state-metrics
    namespace: kube-system
    profile: prod
```

**Local Environment:**  
- Operating System: linux
- Deployment method: helm

**Kubernetes Cluster:**  
- Cloud Provider: minikube
- Kubernetes Version: 
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.9", GitCommit:"2e808b7cb054ee242b68e62455323aa783991f03", GitTreeState:"clean", BuildDate:"2020-01-18T23:24:23Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Anything else we need to know?**  
Render works correctly when dependencies are commented out.

Terminal Dump:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1d05e20]

goroutine 1 [running]:
github.com/devspace-cloud/devspace/pkg/devspace/dependency.(*resolver).resolveDependency(0xc000f87100, 0xc000052104, 0x32, 0xc0016db280, 0x4121400, 0x0, 0x7f672c116d98, 0xc0016e1ad0)
        /Users/runner/runners/2.164.0/work/devspace/devspace/pkg/devspace/dependency/resolver.go:287 +0xa10
github.com/devspace-cloud/devspace/pkg/devspace/dependency.(*resolver).resolveRecursive(0xc000f87100, 0xc000052104, 0x32, 0xc00169f890, 0x21, 0xc001027f90, 0x9, 0x9, 0xc00170c800, 0x29f8820, ...)
        /Users/runner/runners/2.164.0/work/devspace/devspace/pkg/devspace/dependency/resolver.go:169 +0x1e4
github.com/devspace-cloud/devspace/pkg/devspace/dependency.(*resolver).Resolve(0xc000f87100, 0xc00170a900, 0xc00170a960, 0xc00170c840, 0xc0016b39a0, 0x40c828, 0x30)
        /Users/runner/runners/2.164.0/work/devspace/devspace/pkg/devspace/dependency/resolver.go:111 +0xf2
github.com/devspace-cloud/devspace/pkg/devspace/dependency.(*manager).BuildAll(0xc00170c870, 0x0, 0x0, 0x0)
        /Users/runner/runners/2.164.0/work/devspace/devspace/pkg/devspace/dependency/dependency.go:86 +0x9e
github.com/devspace-cloud/devspace/cmd.(*RenderCmd).Run(0xc000ea1b30, 0x2a08d20, 0x411f5f0, 0xc00113d400, 0xc0011346c0, 0x0, 0x2, 0x0, 0x0)
        /Users/runner/runners/2.164.0/work/devspace/devspace/cmd/render.go:120 +0x474
github.com/devspace-cloud/devspace/cmd.NewRenderCmd.func1(0xc00113d400, 0xc0011346c0, 0x0, 0x2, 0x0, 0x0)
        /Users/runner/runners/2.164.0/work/devspace/devspace/cmd/render.go:55 +0x64
github.com/spf13/cobra.(*Command).execute(0xc00113d400, 0xc001134680, 0x2, 0x2, 0xc00113d400, 0xc001134680)
        /Users/runner/runners/2.164.0/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc000f83b80, 0x411f5f0, 0xc000f83b80, 0x0)
        /Users/runner/runners/2.164.0/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/runner/runners/2.164.0/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:864
github.com/devspace-cloud/devspace/cmd.Execute()
        /Users/runner/runners/2.164.0/work/devspace/devspace/cmd/root.go:70 +0xbd
main.main()
        /Users/runner/runners/2.164.0/work/devspace/devspace/main.go:17 +0x3e
```

/kind bug

**Comment by FabianKramm** [comment]
@chasebolt thanks for reporting this! I'll fix this in the next version (which will come out the next days), currently this will only occur if you specify the namespace in the dependency and it should still work with namespace in `devspace dev` & `devspace deploy`.

**Comment by FabianKramm** [comment]
Should be fixed in v4.5.1
